### PR TITLE
fix(core): add ariaLabel to Pagination component

### PR DESCRIPTION
**Pull request by kavya-b**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#3155

#### Please provide a brief summary of this pull request.
Each `nav` element is supposed to have a label [if there are multiple `nav` landmarks in a page.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role#labeling_landmarks). This PR brings in an `ariaLabel` property to address the issue.
Other minor issues like maintaining order of heading element, and clearer label for the 'items per page' buttons are also handled from the documentation side.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

File: libs/core/src/lib/pagination/pagination.component.ts [hunk 1]
```diff
@@ -137,6 +137,9 @@ export class PaginationComponent implements OnChanges, OnInit, OnDestroy {
     @Input()
     pageLabel = 'Page';
 
+    /** Aria label for the navigation element */
+    @Input()
+    ariaLabel = 'Pagination with id ' + this.id;
```
Comment (dimamarksman) [inline libs/core/src/lib/pagination/pagination.component.ts:142]: Why do we use `Pagination with id {id}` as the default one?

Comment (kavya-b) [reply libs/core/src/lib/pagination/pagination.component.ts:142]: We need some label by default when multiple `<nav>`s are used and user does not specify `ariaLabel`. From the issue description and [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role#description),  it is not very clear whether the label has to be unique or not. To be on the safe side, I let it be unique by including the `id`.
image

Comment (dimamarksman) [reply libs/core/src/lib/pagination/pagination.component.ts:142]: just `Pagination`?

Comment (kavya-b) [reply libs/core/src/lib/pagination/pagination.component.ts:142]: just `Pagination` also works when I tried, but if the expectation is that each label must be unique, wouldn't having just `Pagination` reflect the same on all the different `<nav>`s?

Comment (dimamarksman) [reply libs/core/src/lib/pagination/pagination.component.ts:142]: So if pagination must be unique then lib consumer must provide a unique descriptive `aria-label`, this is how I see it.
component.id is not descriptive by itself, it just generates something like `fd-pagination-0` `fd-pagination-1` and so on.
By default we would have `Pagination with id fd-pagination-0`, as for me it looks weird.

Comment (kavya-b) [reply libs/core/src/lib/pagination/pagination.component.ts]: I see, I guess it makes sense that we leave it to consumer to provide unique labels, if necessary. Have kept it as just `Pagination` now, please take a look.

**Review by dimamarksman — APPROVED** [review]

**Comment by puru-hk** [comment]
Looks fine.

**Review by puru-hk — APPROVED** [review]

**Pull request merged**